lms/djangoapps/courseware/masquerade.py

```diff
@@ -154,7 +154,7 @@ def is_masquerading_as_specific_student(user, course_key):  # pylint: disable=in
     Returns whether the user is a staff member masquerading as a specific student.
     """
     course_masquerade = get_course_masquerade(user, course_key)
-    return bool(course_masquerade and course_masquerade.user_name)
+    return bool(course_masquerade and getattr(course_masquerade, 'user_name', None))
```

Comment:
@e-kolpakov Thanks for the suggestion.  I see your point, but I don't really like overwriting `__getattr__()` for this purpose.  Even when logging a warning, this bears the risk of hiding typos in attribute names, so I'm not sure it's worth the added complexity overall.

I've implemented a different approach that makes sure `CourseMasquerade` objects get all required attributes even when unpickled from the session.  Do you think this approach addresses your concern?